.github/workflows/ci.yml

```diff
@@ -103,6 +103,23 @@ jobs:
       - name: Build and Test
         run: python ./build.py --no-check
 
+  format_qsc:
+    name: Format Q# Files
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: "true"
+      - name: Setup rust toolchain
+        uses: ./.github/actions/toolchains/rust
+        with:
+          toolchain: ${{ env.RUST_TOOLCHAIN_VERSION }}
+          components: ${{ env.RUST_TOOLCHAIN_COMPONENTS }}
+      - name: Check Formatting for Libraries
+        run: cargo run --release --bin qsc_formatter -- ./library/ -r
+      - name: Check Formatting for Samples
+        run: cargo run --release --bin qsc_formatter -- ./samples/ -r
```

Comment:
IMO this should be a part of `build.py --check` as well, so that it runs by default locally and so that it's consistent with how/when we run `prettier` and `cargo fmt` today.

Reply:
Oh and if it fails, can it say "please fix the formatting by running `<exact command I can just copy and paste>`"?

Reply:
I added it to the build.py --check step, but I'm not sure about having it print out a custom message if that step fails. I think it always just tells the user that the subprocess failed.

Reply:
Either way. If it makes the script really awkward then I can live without a custom message :)

Reply:
I tried to add an instructive statement to the situations where we want the user to run the formatter with the `--write` option.